### ISSUE TITLE
[denonmarantz] Binding reports 'unknown' in OH3.1.0 and is working in openHAB2.5.12-1

### ISSUE THREAD

**Issue by PatrickGlatz** [openhab/openhab-addons#11502]
Hello maintainers

when setting up denonmarantz binding in OH3.1.0, i am configuring everything like i did in openHAB2.5.12:

Identifier: AVR1912
Label: Denon AVR-1912
Location: Wohnzimmer
ThingType: Denon/Marantz AVR
ZoneCount: 2
IP Adress: 192.168.178.100
Use telnet: NO
Telnet Port: 23
HTTP Port: 80
Polling Interval: 5

I am using an Denon AVR-1912. It is accessible over http (web access).
This configuration works fine for openHAB2.5.12-1 but in OH3.1.0 i am getting directly the following log outputs:

```
2021-11-02 22:16:45.819 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'denonmarantz:avr:AVR1912' changed from UNKNOWN to UNINITIALIZED
2021-11-02 22:16:45.835 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'denonmarantz:avr:AVR1912' changed from UNINITIALIZED to UNINITIALIZED (HANDLER_MISSING_ERROR)
2021-11-02 22:16:45.891 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'denonmarantz:avr:AVR1912' changed from UNINITIALIZED (HANDLER_MISSING_ERROR) to INITIALIZING
2021-11-02 22:16:45.898 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'denonmarantz:avr:AVR1912' changed from INITIALIZING to UNKNOWN
```
Restarting the binding does not fix it nor a reboot of the system does. Issue is also reported here
[openHAB Forum](https://community.openhab.org/t/denon-avr-is-unknown-in-3-1-1-snapshot/123689)

## Expected Behavior
Binding should connect to AVR as in OH2.5.12-1 and should go `online`

## Current Behavior
see description above

## Possible Solution
Unfortunately non

## Steps to Reproduce (for Bugs)
Install binding
Add thing
Configure Thing according to settings on the top

## Context
The issue prevents me from utilizing my Denon within openHAB.

## Your Environment
Denon AVR-1912
raspberryPi 4 8GB
32GB SD
OH3.1.0 via openhabian, 32-bit
openHAB as well as Denon AVR in same subnet 192.168.178.0/24

**Comment by lolodomo** [comment]
Please enable the DEBUG logs of the binding. Restart the binding and show us the logs.

**Comment by PatrickGlatz** [comment]
Hi @lolodomo 

please find the TRACE log below. Thanks for your support! Really appreciate it. 

```
2021-11-14 19:50:31.186 [TRACE] [ector.http.DenonMarantzHttpConnector] - Refreshing Denon status
2021-11-14 19:50:31.187 [TRACE] [ector.http.DenonMarantzHttpConnector] - Refreshing URL: http://192.168.144.101:80/goform/formMainZone_MainZoneXml.xml
2021-11-14 19:50:31.229 [TRACE] [ector.http.DenonMarantzHttpConnector] - result of getDocument for uri 'http://192.168.144.101:80/goform/formMainZone_MainZoneXml.xml':
<?xml version="1.0" encoding="utf-8" ?>
<item>
<Power><value>ON</value></Power>
<ZonePower><value>ON</value></ZonePower>
<InputFuncList>
<value>SOURCE</value>
<value>TUNER</value>
<value>CD</value>
<value>NET/USB</value>
<value>DVD</value>
<value>BD</value>
<value>TV</value>
<value>SAT/CBL</value>
<value>DOCK</value>
<value>GAME2</value>
<value>GAME</value>
<value>V.AUX</value>
</InputFuncList>
<RenameSource>
<value></value>
<value>Radio   </value>
<value>Vinyl   </value>
<value>NetRadio</value>
<value>PS4     </value>
<value>Ext.HDMI</value>
<value>TV      </value>
<value>VINYL   </value>
<value>BT      </value>
<value>Apple TV</value>
<value>openELEC</value>
<value>V.AUX   </value>
</RenameSource>
<RenameZone><value>MAIN ZONE 
</value></RenameZone>
<SourceDelete>
<value>DEL</value>
<value>USE</value>
<value>USE</value>
<value>USE</value>
<value>USE</value>
<value>USE</value>
<value>USE</value>
<value>DEL</value>
<value>USE</value>
<value>USE</value>
<value>USE</value>
<value>DEL</value>
</SourceDelete>
<TopMenuLink><value>ON</value></TopMenuLink>
<ModelId><value>1</value></ModelId>
<SalesArea><value>1</value></SalesArea>
<InputFuncSelect><value>NET/USB</value></InputFuncSelect>
<NetFuncSelect><value>NET/USB</value></NetFuncSelect>
<InputFuncSelectMain><value>NET/USB</value></InputFuncSelectMain>
<selectSurround><value>MULTI CH STEREO                </value></selectSurround>
<SurroundLists>
<value index=''  table='' param=''/>
<value index='DOLBY DIGITAL'  table='Standard(Dolby)' param=''/>
<value index='DTS SURROUND'  table='Standard(DTS)' param=''/>
<value index='DIRECT'  table='Direct' param=''/>
<value index='PURE DIRECT'  table='Pure Direct' param=''/>
<value index='STEREO'  table='Stereo' param=''/>
<value index='MCH STEREO'  table='Multi CH Stereo' param=''/>
<value index='VIRTUAL'  table='Virtual' param=''/>
<value index='ROCK ARENA'  table='Rock Arena' param=''/>
<value index='JAZZ CLUB'  table='Jazz Club' param=''/>
<value index='MONO MOVIE'  table='Mono Movie' param=''/>
<value index='VIDEO GAME'  table='Video Game' param=''/>
<value index='MATRIX'  table='Matrix' param=''/>
</SurroundLists>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<MasterVolume><value>-54.5</value></MasterVolume>
<Mute><value>off</value></Mute>
<RemoteMaintenance><value>OFF</value></RemoteMaintenance>
<GameSourceDisplay><value>TRUE</value></GameSourceDisplay>
<LastfmDisplay><value>FALSE</value></LastfmDisplay>
<SubwooferDisplay><value>FALSE</value></SubwooferDisplay>
<Zone2VolDisp><value>TRUE</value></Zone2VolDisp>
</item>
2021-11-14 19:50:31.234 [DEBUG] [ector.http.DenonMarantzHttpConnector] - Failed to unmarshal xml document: null
2021-11-14 19:50:31.235 [TRACE] [ector.http.DenonMarantzHttpConnector] - Refreshing URL: http://192.168.144.101:80/goform/formMainZone_MainZoneXmlStatus.xml
2021-11-14 19:50:31.302 [TRACE] [ector.http.DenonMarantzHttpConnector] - result of getDocument for uri 'http://192.168.144.101:80/goform/formMainZone_MainZoneXmlStatus.xml':
<?xml version="1.0" encoding="utf-8" ?>
<item>
<Zone><value>MAIN ZONE 
</value></Zone>
<Power><value>ON</value></Power>
<InputFuncList>
<value>SOURCE</value>
<value>TUNER</value>
<value>CD</value>
<value>NET/USB</value>
<value>DVD</value>
<value>BD</value>
<value>TV</value>
<value>SAT/CBL</value>
<value>DOCK</value>
<value>GAME2</value>
<value>GAME</value>
<value>V.AUX</value>
</InputFuncList>
<RenameSource>
<value></value>
<value>Radio   </value>
<value>Vinyl   </value>
<value>NetRadio</value>
<value>PS4     </value>
<value>Ext.HDMI</value>
<value>TV      </value>
<value>VINYL   </value>
<value>BT      </value>
<value>Apple TV</value>
<value>openELEC</value>
<value>V.AUX   </value>
</RenameSource>
<InputFuncSelect><value>NET/USB</value></InputFuncSelect>
<VolumeDisplay><value>Relative</value></VolumeDisplay>
<RestorerMode><value>----</value></RestorerMode>
<SurrMode><value>MCH STEREO</value></SurrMode>
<MasterVolume><value>-54.5</value></MasterVolume>
<Mute><value>off</value></Mute>
<Model><value>19XX E2</value></Model>
</item>
2021-11-14 19:50:31.309 [DEBUG] [ector.http.DenonMarantzHttpConnector] - Failed to unmarshal xml document: null
2021-11-14 19:50:31.310 [TRACE] [ector.http.DenonMarantzHttpConnector] - Refreshing URL: http://192.168.144.101:80/goform/AppCommand.xml
2021-11-14 19:50:31.338 [DEBUG] [ector.http.DenonMarantzHttpConnector] - Encoding error in post
javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}cmd>,<{}list>,<{}rx>,<{}text>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.startElement(SAXConnector.java:137) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:510) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:374) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl$NSContentDriver.scanRootElementHook(XMLNSDocumentScannerImpl.java:613) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:3063) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:836) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112) ~[?:?]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141) ~[?:?]
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216) ~[?:?]
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635) ~[?:?]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:228) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:199) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:140) ~[jakarta.xml.bind-api-2.3.3.jar:2.3.3]
	at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:179) ~[jakarta.xml.bind-api-2.3.3.jar:2.3.3]
	at org.openhab.binding.denonmarantz.internal.connector.http.DenonMarantzHttpConnector.postDocument(DenonMarantzHttpConnector.java:348) ~[?:?]
	at org.openhab.binding.denonmarantz.internal.connector.http.DenonMarantzHttpConnector.updateDisplayInfo(DenonMarantzHttpConnector.java:262) ~[?:?]
	at org.openhab.binding.denonmarantz.internal.connector.http.DenonMarantzHttpConnector.refreshHttpProperties(DenonMarantzHttpConnector.java:298) ~[?:?]
	at org.openhab.binding.denonmarantz.internal.connector.http.DenonMarantzHttpConnector.lambda$0(DenonMarantzHttpConnector.java:126) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
```

**Comment by lolodomo** [comment]
Is it a different issue than #10962  ? 

It looks like a problem with jaxb.I don't know what jaxb version was used in OH 2.5.
Maybe a refactoring to parse XML responses without jaxb would help. 

@jwveldhuis : are you still there to maintain this binding?

**Comment by lolodomo** [comment]
Here is what is recommended for XML oarsing in the coding guidelines. 
https://www.openhab.org/docs/developer/guidelines.html#default-libraries
com.thoughtworks.xstream

**Comment by lolodomo** [comment]
@clinique : I see one of your bindings (xmltv) is using jaxb. Can you confirm it is still working in OH 3.x with Java 11 and if it is, any idea what could wrong in the denonmarantz binding?

**Comment by clinique** [comment]
Yes, xmltv is working in oh 3.2. From a satellite view, the error seems to come from unmarshalling, maybe the XML message has changed ?

**Comment by PatrickGlatz** [comment]
Hi @lolodomo ,

thanks for your efforts. Regarding your link to the other issue: I can confirm that I do have the same behaviour. Controlling is possible, but the status is not reported back (although the passed back xml shows everything correct).

is there anything I can do from a user side?

**Comment by lolodomo** [comment]
> maybe the XML message has changed ?

Probably not as it works on 2.5.
Except if there were changes in 3.x version of the binding.

**Comment by lolodomo** [comment]
> is there anything I can do from a user side?

You need help from the binding author / maintainer or a new volunteer owning such device.

**Comment by lolodomo** [comment]
#10614 could have impacted something.

**Comment by PatrickGlatz** [comment]
@wborn Can you help us with this issue?

**Comment by PatrickGlatz** [comment]
How can I find out who is the binding author and how to contact him?

**Comment by lolodomo** [comment]
> How can I find out who is the binding author and how to contact him?

The author/maintainer of each binding is defined in the file CODEOWNERS presnet at the root of this repo.
I already pinged him as you can see above.

**Comment by pieterbuts** [comment]
I found the solution and proved it working on my raspberry pi running openHABian 3.1.0 connecting to my Denon receiver.

See https://github.com/pieterbuts/openhab-addons/commit/6bc39d8d98c95c66b01f12a3d23b78e9bdd58800 for details.

As you can see I created a fork and a feature branch, but I don't know how to get it integrated in the 3.1.x addons. So I would appreciate some help with that.

**Comment by lolodomo** [comment]
Fixed by #11766
